ui/cyoa-renderer/src/main/res/layout/cyoa_page_content.xml

```diff
@@ -1,26 +1,44 @@
 <?xml version="1.0" encoding="utf-8"?>
-<layout xmlns:android="http://schemas.android.com/apk/res/android">
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
     <data>
         <import type="androidx.lifecycle.LiveData" />
+        <import type="org.cru.godtools.tool.model.page.PageKt" />
         <import type="org.cru.godtools.base.tool.ui.util.BaseUtilsKt" />
 
         <variable name="page" type="org.cru.godtools.tool.model.page.ContentPage" />
         <variable name="controller" type="org.cru.godtools.tool.cyoa.ui.controller.ContentPageController" />
         <variable name="contentInsets" type="LiveData&lt;androidx.core.graphics.Insets&gt;" />
     </data>
 
-    <org.ccci.gto.android.common.widget.HackyNestedScrollView
-        android:id="@+id/initial_page_content"
-        android:layout_marginTop="@{contentInsets.top}"
+    <FrameLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:fillViewport="true"
-        android:layoutDirection="@{BaseUtilsKt.getLayoutDirection(page)}">
+        android:background="@{PageKt.getBackgroundColor(page)}">
 
-        <LinearLayout
-            android:id="@+id/content"
+        <org.cru.godtools.base.tool.widget.SimpleScaledPicassoImageView
```

Comment:
So ```FrameLayout``` handles the background color while ```SimpleScaledPicassoImageView``` handles the background image if there is one. That way we can have a background color of, let's say, blue, and then a scaled background image and be able to see both at the same time. I guess you could technically scale the image to cover the whole screen if you wanted right?

Reply:
yes, we want to layer our 2 background layers here, and then put the content on top of it. `FrameLayout` is a very lightweight android `ViewGroup`. it can have multiple children that are drawn on top of each other from first to last. so we draw any defined background image as the first child of the FrameLayout, and then the content stack is drawn on top of the background as the second child in the `FrameLayout`.

once we have that to get the backgroundColor behind the backgroundImage we attach it to the `FrameLayout` as the wrapper element.